dotnet/src/Agents/Core/Chat/SequentialSelectionStrategy.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.SemanticKernel.Agents.Chat;
+
+/// <summary>
+/// Round-robin turn-taking strategy.
+/// </summary>
+public sealed class SequentialSelectionStrategy : SelectionStrategy
+{
+    private int _index = 0;
```

Comment:
The only way to reset the index at the moment is to create a new instance of a strategy, which would likely require instantiating the entire group chat, correct? I wonder if we can pass the index/turn from the agent group chat and use it to decide on the next agent. This way, the strategy wouldn't need to be reset every time chat users interact with the same chat.

Reply:
Yes, I think exposing a way to reset doesn't conflict with any contract here and can provide more control.

(Will visit as part of the default strategies update)

Reply:
I explored exposing a `CurrentAgent` property along with `Reset` semantics, but quickly realized that the strategy (rightly) doesn't maintain agent state...the agent collection is provided as a parameter as this is the core responsibility of `AgentGroupChat`.

Due to this same consideration, even exposing an `Index` property become problematic now that its incrementing to MaxValue with the mod operator. (butterfly effect)

Anyway, I've gone ahead and just settled on a straight `Reset()` semantic.  In the case where turn taking is in order (this case), I think its justifiable to express the desire to "start over".  How this considerationprojects into other strategies will be interesting to explore and something to keep in mind.